api/types/src/transaction.rs

```diff
@@ -586,13 +570,278 @@ pub struct BlockMetadataTransaction {
+            ValidatorTransaction::ObservedJwkUpdate(_) => "validator_transaction__observed_jwk_update",
+            ValidatorTransaction::DkgResult(_) => "validator_transaction__dkg_result",
+        }
+    }
+
+    pub fn transaction_info(&self) -> &TransactionInfo {
+        match self {
+            ValidatorTransaction::ObservedJwkUpdate(t) => &t.info,
+            ValidatorTransaction::DkgResult(t) => &t.info,
+        }
+    }
+
+    pub fn transaction_info_mut(&mut self) -> &mut TransactionInfo {
+        match self {
+            ValidatorTransaction::ObservedJwkUpdate(t) => &mut t.info,
+            ValidatorTransaction::DkgResult(t) => &mut t.info,
+        }
+    }
+
+    pub fn timestamp(&self) -> U64 {
+        match self {
+            ValidatorTransaction::ObservedJwkUpdate(t) => t.timestamp,
+            ValidatorTransaction::DkgResult(t) => t.timestamp,
+        }
+    }
+
+    pub fn from_internal_repr(
+        internal: aptos_types::validator_txn::ValidatorTransaction,
+        info: TransactionInfo,
+        events: Vec<Event>,
+        timestamp: u64,
+    ) -> Self {
+        match internal {
+            aptos_types::validator_txn::ValidatorTransaction::DKGResult(dkg_transcript) => {
+                Self::DkgResult(DKGResultTransaction {
+                    info,
+                    events,
+                    timestamp: U64::from(timestamp),
+                    dkg_transcript: ExportedDKGTranscript::from_internal_repr(dkg_transcript),
+                })
+            },
+            aptos_types::validator_txn::ValidatorTransaction::ObservedJWKUpdate(quorum_certified_update) => {
+                Self::ObservedJwkUpdate(JWKUpdateTransaction {
+                    info,
+                    events,
+                    timestamp: U64::from(timestamp),
+                    quorum_certified_update: ExportedQuorumCertifiedUpdate::from_internal_repr(quorum_certified_update),
+                })
+            },
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct JWKUpdateTransaction {
+    #[serde(flatten)]
+    #[oai(flatten)]
+    pub info: TransactionInfo,
+    pub events: Vec<Event>,
+    pub timestamp: U64,
+    pub quorum_certified_update: ExportedQuorumCertifiedUpdate,
+}
+
+/// A more API-friendly representation of the on-chain `aptos_types::jwks::QuorumCertifiedUpdate`.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct ExportedQuorumCertifiedUpdate {
+    pub update: ExportedProviderJWKs,
+    pub multi_sig: ExportedAggregateSignature,
+}
+
+impl ExportedQuorumCertifiedUpdate {
+    pub fn from_internal_repr(internal: QuorumCertifiedUpdate) -> Self {
+        let QuorumCertifiedUpdate { update, multi_sig } = internal;
+        Self {
+            update: ExportedProviderJWKs::from_internal_repr(update),
+            multi_sig: ExportedAggregateSignature::from_internal_repr(multi_sig),
+        }
+    }
+}
```

Comment:
This is usually best represented in Rust as a straight implements From

Reply:
i found them hard to locate, and the only benefit is `.into()`?

Reply:
It's more appropriately sharable across.

So, you can do both From and Into at any place.  Helps with other things automatically as well.